### ISSUE TITLE
Lockbox and ActionText

### ISSUE THREAD

**Issue by phildionne** [ankane/lockbox#22]
Using Rails, I wanted to encrypt ActionText records and thought I would share how I did it here. @ankane you're welcome to include it in the official documentation or leave it here for posterity's sake.

# ActionText Encrypted

1. Add column
2. Migrate
3. Write to encrypted column
4. Remove unencrypted column

## Add column

```ruby
# db/migrate/...
class AddBodyCiphertextToActionTextRichTexts < ActiveRecord::Migration[6.0]
  def change
    add_column :action_text_rich_texts, :body_ciphertext, :text
  end
end

# initializers/patches.rb
module Patches
  module ActionText
    module EncryptedBody
      extend ActiveSupport::Concern

      included do
        encrypts :body, migrating: true
      end
    end
  end
end

ActiveSupport.on_load(:action_text_rich_text) do
  ActionText::RichText.include(Patches::ActionText::EncryptedBody)
end

ActiveSupport.on_load(:action_text_content) do
  module Patches
    module ActionText
      module IO
        extend ActiveSupport::Concern

        def read
          to_s
        end
      end
    end
  end

  ActionText::Content.include(Patches::ActionText::IO)
end
```

## Migrate

```ruby
Lockbox.migrate(ActionText::RichText)
```

## Write to encrypted column

```ruby
# initializers/patches.rb
module Patches
  module ActionText
    module EncryptedBody
      extend ActiveSupport::Concern

      included do
        encrypts :body
        self.ignored_columns = ["body"]
      end
    end
  end
end

ActiveSupport.on_load(:action_text_rich_text) do
  ActionText::RichText.include(Patches::ActionText::EncryptedBody)
end
```

## Remove unencrypted column

```ruby
class RemoveBodyFromActionTextRichTexts < ActiveRecord::Migration[6.0]
  def change
    remove_column :action_text_rich_texts, :body, :text
  end
end
```

```ruby
# initializers/patches.rb
module Patches
  module ActionText
    module EncryptedBody
      extend ActiveSupport::Concern

      included do
        encrypts :body
      end
    end
  end
end

ActiveSupport.on_load(:action_text_rich_text) do
  ActionText::RichText.include(Patches::ActionText::EncryptedBody)
end
```

**Comment by ankane** [comment]
Awesome, thanks for sharing @phildionne! I haven't tried Action Text yet, but will give this a shot when I do and decide how to proceed. It may make sense to build this into Lockbox.

**Comment by ankane** [comment]
Need to do some more testing, but started Action Text integration here: https://github.com/ankane/lockbox/compare/action_text

**Comment by ankane** [comment]
Just released 0.4.1 with official support for Action Text. The method name was changed slightly to:

```ruby
Lockbox.encrypts_action_text_body
```

Docs: https://github.com/ankane/lockbox#action-text

Thanks for sharing your approach @phildionne! I was able to get it running without the `ActionText::Content` patch, but let me know if I missed something.
